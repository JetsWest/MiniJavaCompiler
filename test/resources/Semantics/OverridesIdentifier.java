class SomeClass {
    public static void main(String[] args) {
        System.out.println(0);
    }
}

class First {
    public int first() {
        return 0;
    }
}

class Second {
    public int second() {
        return 0;
    }
}

class A {
    public First foo() {
        return new First();
    }
}

class B extends A {
    public First foo() {
        return new First();
    }
}

class C extends A {
    public Second foo() {
        return new Second();
    }
}

class D extends A {
    public First foo(int x) {
        return new First();
    }
}

class E {
    public First foo(int x) {
        return new First();
    }
}

class F extends E {
    public First foo(boolean x) {
        return new First();
    }
}