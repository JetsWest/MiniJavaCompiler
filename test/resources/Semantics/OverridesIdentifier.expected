(Line 32) Overrode methods must have same return type.
(Line 38) Overrode methods must have same number of parameters.
(Line 50) Overrode methods must have same parameter types.
